### ISSUE TITLE
BIE Reuse for Score 2.0

### ISSUE THREAD

**Issue by kbserm** [OAGi/Score#859]
This is to port code for #821 to 2.0

**Comment by kbserm** [comment]
@asdimitriadis Make sure to test that BIE in one release cannot reuse a BIE in another release.

**Comment by kbserm** [comment]
Looks like there is a different in the logic when remove result b/w 1.x and this 2.0. When I removed reuse, the BIE content comes back. 

Below is quoted from #858. It looks like in 2.0, the replaced ASBIEP does not get deleted right away according to this quote when the reuse is applied. 

"We also talked about potential problem if the user reuse and remove reuse so quick that the deletion hasn't occurred or finished. But this is not a problem b/c @hakjuoh made sure that the replaced ASBIEP record got deleted b/f the rest of the process. It is just that its descendants are deleted later."
